api/src/routes/users.ts

```diff
@@ -98,4 +103,46 @@ router.delete("/connections/:id", routeProtector, [ param("id").not().isEmpty().
   }
 });
 
+
+router.patch("/:id", routeProtector, [body("firstName").trim().escape(),
+body("lastName").trim().escape(),
+body("avatar").isURL().trim(),
+body("jobTitle").trim().escape(),
+param("id").not().isEmpty().trim().escape()],
+  async (req: any, res: Response) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).send({ errors: errors.array() });
+    }
+
+    if (req.params.id !== "me") {
+      return res.status(400).send({errors: [{
+        "location": "Patch profile update",
+        "msg": `Bad request`,
+        "param": req.params.id
+      }]});
+    }
+    const profileUpdateRequest: IProfileData = {
+      firstName: req.body.firstName,
+      lastName: req.body.lastName,
+      jobTitle: req.body.jobTitle,
+      avatarUrl: req.body.avatar
+    };
+
+    try {
+      await req.user.updateUserProfile(profileUpdateRequest);
+      res.status(200).send({ firstName: req.user.firstName,
```

Comment:
This could be dryer... {...profileUpdateRequest, email: decrypt(req.user.auth.email) }

Reply:
Ah yeah, just noticed avatarURL to avatar, not as much a saving then! :joy:

Reply:
> Looking good so far, I'm just working through it by adjusting the front end feature/6-user-profile branch...
> One find is that the get route doesn't return the job title, in src/db/utils/get-profile-by-id

Ty for catching. updated!